modules/backup-s3/client.go

```diff
@@ -146,28 +174,46 @@ func (s *s3Client) PutObject(ctx context.Context, backupID, key string, byes []b
 	return nil
 }
 
-func (s *s3Client) Initialize(ctx context.Context, backupID string) error {
+func (s *s3Client) Initialize(ctx context.Context, backupID, overrideBucket, overridePath string) error {
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to create temporary client")
+	}
+
 	key := "access-check"
 
-	if err := s.PutObject(ctx, backupID, key, []byte("")); err != nil {
+	if err := s.PutObject(ctx, backupID, key, overrideBucket, overridePath, []byte("")); err != nil {
 		return errors.Wrap(err, "failed to access-check s3 backup module")
 	}
 
 	objectName := s.makeObjectName(backupID, key)
 	opt := minio.RemoveObjectOptions{}
-	if err := s.client.RemoveObject(ctx, s.config.Bucket, objectName, opt); err != nil {
+	if err := client.RemoveObject(ctx, s.config.Bucket, objectName, opt); err != nil {
 		return errors.Wrap(err, "failed to remove access-check s3 backup module")
 	}
 
 	return nil
 }
 
 // WriteFile downloads contents of an object to a local file destPath
-func (s *s3Client) WriteToFile(ctx context.Context, backupID, key, destPath string) error {
-	object := s.makeObjectName(backupID, key)
-	err := s.client.FGetObject(ctx, s.config.Bucket, object, destPath, minio.GetObjectOptions{})
+func (s *s3Client) WriteToFile(ctx context.Context, backupID, key, destPath, overrideBucket, overridePath string) error {
+	client, err := s.getClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "WriteToFile")
```

Comment:
suggestion: maybe `cannot get client` message here would be better?

Reply:
done